demoapp-ios-swift/MasterViewController.swift

```diff
@@ -11,29 +11,23 @@ import UIKit
 class MasterViewController: UITableViewController {
     
     var detailViewController: DetailViewController? = nil
-    var objects: [Any] = ["Build", "Test", "Distribute", "Crash", "Analytics", "Push"]
-    
-    // construct Data Model array, manually added for now
+    // construct Data Model array
     var beacons: [Beacon] = []
     
-    required init(coder aDecoder: NSCoder) {
-        super.init(coder: aDecoder)!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view
+        // Here, we manually add our beacons to the Data Model
         self.beacons.append(Beacon(name: "Build", type: BeaconType.Build))
         self.beacons.append(Beacon(name: "Test", type: BeaconType.Test))
         self.beacons.append(Beacon(name: "Distribute", type: BeaconType.Distribute))
         self.beacons.append(Beacon(name: "Crash", type: BeaconType.Crash))
         self.beacons.append(Beacon(name: "Analytics", type: BeaconType.Analytics))
         self.beacons.append(Beacon(name: "Push", type: BeaconType.Push))
-    }
-    
-    
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
         
         if let split = splitViewController {
             let controllers = split.viewControllers
-            detailViewController = (controllers[controllers.count-1] as! UINavigationController).topViewController as? DetailViewController
+            self.detailViewController = (controllers[controllers.count-1] as! UINavigationController).topViewController as? DetailViewController
```

Comment:
nit: `controllers[controllers.count - 1`
can you check what does the styling tool enforce?